File: backend/TweetMicroApi/TweetMicroApi/Models/Tweet.cs
namespace TweetMicroApi.Models
{
    public class Tweet
    {
        public int Id { get; set; }
        public string Username { get; set; }
        public string Content { get; set; }
        public DateTime? Timestamp { get; set; }
        public int Likes { get; set; }
        public int Retweets { get; set; }
    }
}

File: twitter_app_with_auth/lib/models/tweet.dart
class Tweet {
  Tweet(this.id, this.username, this.content, this.timestamp, this.likes,
      this.retweets);
  int id;
  String username;
  String content;
  String timestamp;
  int likes;
  int retweets;

  Map<String, dynamic> toJson() {
    Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['username'] = username;
    data['content'] = content;
    data['likes'] = likes;
    data['retweets'] = retweets;
    return data;
  }

  static Tweet fromJson(Map<String, dynamic> data) {
    return Tweet(
      data['id'],
      data['username'],
      data['content'],
      data['timestamp'].toString(),
      data['likes'],
      data['retweets'],
    );
  }
}
